core/src/main/java/com/novoda/noplayer/exoplayer/NoPlayerExoPlayerCreator.java

```diff
@@ -0,0 +1,88 @@
+package com.novoda.noplayer.exoplayer;
+
+import android.content.Context;
+import android.os.Handler;
+
+import com.google.android.exoplayer2.trackselection.DefaultTrackSelector;
+import com.google.android.exoplayer2.trackselection.FixedTrackSelection;
+import com.google.android.exoplayer2.upstream.DefaultDataSourceFactory;
+import com.novoda.noplayer.Heart;
+import com.novoda.noplayer.LoadTimeout;
+import com.novoda.noplayer.Player;
+import com.novoda.noplayer.PlayerListenersHolder;
+import com.novoda.noplayer.SystemClock;
+import com.novoda.noplayer.drm.DrmSessionCreator;
+import com.novoda.noplayer.exoplayer.forwarder.ExoPlayerForwarder;
+import com.novoda.noplayer.exoplayer.mediasource.ExoPlayerAudioTrackSelector;
+import com.novoda.noplayer.exoplayer.mediasource.ExoPlayerSubtitleTrackSelector;
+import com.novoda.noplayer.exoplayer.mediasource.ExoPlayerTrackSelector;
+import com.novoda.noplayer.exoplayer.mediasource.MediaSourceFactory;
+
+public class NoPlayerExoPlayerCreator {
+
+    private final InternalCreator internalCreator;
```

Comment:
I'm not sure we gain much doing this (apart from creating a whole class). You can just hold a reference to the handler and then use a static method just use a local create method. We then reduce indirection.